### ISSUE TITLE
v2.2.0 freezes my IntelliJ IDEA 2020.1

### ISSUE THREAD

**Issue by seniorjoinu** [intellij-solidity/intellij-solidity#198]
I have a big solidity project (like 5k lines). After 2.2.0 update I'm no longer able to edit any of my `.sol` files - IDE just freezes when I start typing. It may unfreeze after some time, but then it freezes again. 

And also task manager shows increased resources consumption for IntelliJ.

I'm on Windows 10.

**Comment by SerCeMan** [comment]
Hey, @seniorjoinu! Could you attach some debugging information? It's hard to tell what could cause this without a stack or profile. You can find detailed instructions on how to collect some debugging information here: https://intellij-support.jetbrains.com/hc/en-us/articles/207241235-Reporting-performance-problems.

**Comment by seniorjoinu** [comment]
Hey there! I did `jstack` when the IDE was frozen. Here are logs
https://pastebin.com/SswDm2ws

**Comment by SerCeMan** [comment]
@seniorjoinu, for some reason the jstack output for the main application thread seems to be truncated. Could take a dump with `jstack -l -e` arguments? I'm not 100% sure but it might prevent the truncation. 

Additionally, it seems like such a long recursion could result in a StackOverflowError. Could you check the logs of you IDE? If there's an exception, it'll shed some light on the issue.

**Comment by seniorjoinu** [comment]
Looks like there is no `-e` flag for `jstack` that is bundled with my IDEA 
```
C:\Users\Joinu>jstack -l -e 4100 > dump.txt                                                                             
Usage:                                                                                                                      
jstack [-l] <pid>                                                                                                           (to connect to running process
jstack -F [-m] [-l] <pid>                                                                                                   (to connect to a hung process
jstack [-m] [-l] <executable> <core>                                                                                        (to connect to a core file
jstack [-m] [-l] [server_id@]<remote server IP or hostname>                                                                 (to connect to a remote debug server
Options:                                                                                                                    
-F  to force a thread dump. Use when <pid> does not respond (process is hung)
-m  to print both java and native frames (mixed mode)
-l  long listing. Prints additional information about help-h or -help to print this help message 
```

**Comment by SerCeMan** [comment]
@seniorjoinu Thanks! One of the threaddumps from the logs contains the full thread dump which shows where the recursion starts: 
```
11621         at kotlin.sequences.FlatteningSequence$iterator$1.next(Sequences.kt:261)
11622         at kotlin.sequences.FlatteningSequence$iterator$1.next(Sequences.kt:261)
11623         at kotlin.sequences.FlatteningSequence$iterator$1.next(Sequences.kt:261)
11624         at kotlin.sequences.FlatteningSequence$iterator$1.next(Sequences.kt:261)
11625         at kotlin.sequences.DistinctIterator.computeNext(Sequences.kt:524)
11626         at kotlin.collections.AbstractIterator.tryToComputeNext(AbstractIterator.kt:42)
11627         at kotlin.collections.AbstractIterator.hasNext(AbstractIterator.kt:29)
11628         at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator(Sequences.kt:269)
11629         at kotlin.sequences.FlatteningSequence$iterator$1.hasNext(Sequences.kt:265)
11630         at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:131)
11631         at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:156)
11632         at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:722)
11633         at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:752)
11634         at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:743)
11635         at me.serce.solidity.lang.resolve.SolResolver.resolveVarLiteral(engine.kt:167)
11636         at me.serce.solidity.lang.resolve.ref.SolFunctionCallReference.resolveFunctionCall(refs.kt:91)
11637         at me.serce.solidity.lang.resolve.ref.SolFunctionCallReference.resolveFunctionCallAndFilter(refs.kt:199)
11638         at me.serce.solidity.lang.resolve.ref.SolFunctionCallReference.multiResolve(refs.kt:194)
11639         at me.serce.solidity.lang.resolve.SolResolver.resolveVarLiteralReference(engine.kt:132)
11640         at me.serce.solidity.lang.resolve.ref.SolVarLiteralReference.multiResolve(refs.kt:20)
11641         at me.serce.solidity.lang.resolve.ref.SolVarLiteralReference.multiResolve(refs.kt:19)
11642         at me.serce.solidity.lang.resolve.ref.SolReferenceBase$multiResolve$1.resolve(SolReferenceBase.kt:21)
11643         at me.serce.solidity.lang.resolve.ref.SolReferenceBase$multiResolve$1.resolve(SolReferenceBase.kt:14)
```

I'll take a look into this.

**Comment by SerCeMan** [comment]
@0v1se If I understand correctly, this could be caused by one of the recent changes with resolution. Do you know off the top of your head what could result in this? If not, I'll try to dig in and see if I can reproduce it somehow.

**Comment by shark0der** [comment]
Hey! I have a similar problem. In my case the ide is not blocked but it consumes 100% CPU. To trigger it I just open .sol files. As soon as I close them - everything goes back to normal. Interestingly enough - one file gets eats up 100% cpu on a single thread, others all threads. One contract does not trigger it at all. Here're the details:

Repo: `https://github.com/NexusMutual/smart-contracts/`
Branch: `feature/pooled-staking-integration`

100% CPU on all threads files:
- ClaimsData.sol
- ClaimsReward.sol
- Claims.sol
- Governance.sol

100% CPU on a single thread:
- PooledStaking.sol

No issues:
- Quotation.sol

Would really appreciate finding a fix for this because I'll have to switch IDE for the time being (it's almost unusable for now).

**Comment by SerCeMan** [comment]
Hey, folks! I believe the issue should be fixed in `2.2.1`, the version is currently under review in the JB marketplace, in the meantime, you can get the new latest build from [here](https://github.com/intellij-solidity/intellij-solidity/releases/tag/2.2.1).
